File: frontend/src/lib/rent.ts
import { apiFetch } from "@/lib/api";
import type { ChargeInfo } from "@/lib/charges";

export interface LeaseChargeGroup {
  lease_id: string;
  property_address: string;
  tenant_name: string;
  total: number;
  oldest_due: string;
  charges: ChargeInfo[];
}

export interface RentSummary {
  overdue: LeaseChargeGroup[];
  upcoming: LeaseChargeGroup[];
}

export function getRentSummary() {
  return apiFetch<RentSummary>("/api/v1/rent/summary");
}

File: backend/app/schemas/rent.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel

from app.schemas.charge import ChargeInfo


class LeaseChargeGroup(BaseModel):
    """One lease's unsettled charges in a single bucket."""

    lease_id: uuid.UUID
    property_address: str
    tenant_name: str
    total: Decimal
    oldest_due: date
    charges: list[ChargeInfo]


class RentSummary(BaseModel):
    overdue: list[LeaseChargeGroup]
    upcoming: list[LeaseChargeGroup]
